File: mobile/library/swift/test/MockEnvoyHTTPStream.swift
import Envoy
import EnvoyEngine
import Foundation

final class MockEnvoyHTTPStream: EnvoyHTTPStream {
  var onHeaders: (([String: [String]], Bool) -> Void)?
  var onData: ((Data, Bool) -> Void)?
  var onTrailers: (([String: [String]]) -> Void)?

  init(handle: Int, callbacks: EnvoyHTTPCallbacks) {}

  func sendHeaders(_ headers: [String: [String]], close: Bool) {
    self.onHeaders?(headers, close)
  }

  func send(_ data: Data, close: Bool) {
    self.onData?(data, close)
  }

  func sendTrailers(_ trailers: [String: [String]]) {
    self.onTrailers?(trailers)
  }

  func cancel() -> Int32 { return 0 }

  func cleanUp() {}
}

File: mobile/library/kotlin/io/envoyproxy/envoymobile/mocks/MockEnvoyHTTPStream.kt
package io.envoyproxy.envoymobile

import io.envoyproxy.envoymobile.engine.EnvoyHTTPStream
import io.envoyproxy.envoymobile.engine.types.EnvoyHTTPCallbacks
import java.nio.ByteBuffer

/**
 * Internal no-op mock implementation of the engine's `EnvoyHTTPStream`.
 *
 * @param callbacks Callbacks associated with the stream.
 */
internal class MockEnvoyHTTPStream(
  val callbacks: EnvoyHTTPCallbacks,
  val explicitFlowControl: Boolean
) : EnvoyHTTPStream(0, callbacks, explicitFlowControl) {
  override fun sendHeaders(headers: MutableMap<String, MutableList<String>>?, endStream: Boolean) {}

  override fun sendData(data: ByteBuffer?, endStream: Boolean) {}

  override fun readData(byteCount: Long) {}

  override fun sendTrailers(trailers: MutableMap<String, MutableList<String>>?) {}

  override fun cancel(): Int {
    return 0
  }
}
